### PR TITLE
SUPPORTESC-148 fix(core): stashFile no longer throws 'source.on' error when a request that uses await is passed in

### PR DESCRIPTION
**Pull request by nicolemichellethomas**
This Pull Request fixes [this issue](https://github.com/zapier/zapier-platform/issues/169)

Previously, when await was used on a request that was passed to z.stashFile like so:

```
const x = await z.request(...);
return z.stashFile(x);
```

formData would throw the `source.on is not a function`. This PR fixes that issue. 

Jira: https://zapierorg.atlassian.net/browse/SUPPORTESC-148

### CODE REVIEW

**Comment by xavdid** [comment]
@nicolemichellethomas thanks for this PR! I'll dig into this in earnest tomorrow (unless someone else beats me to it). In the meantime, can you link your jira card and provide a little context in the PR description? End-users don't have jira access, so being able to read the PR helps them understand changes for the code they use. 

Sounds like this one fixing an issue where an error is thrown, but shouldn't be?

**Comment by nicolemichellethomas** [comment]
@xavdid You bet, sorry about that! Wasn't sure about the protocol 😅 Let me know if you have any questions or need anything else!

File: packages/core/src/tools/create-file-stasher.js [hunk 1]
```diff
@@ -105,70 +105,69 @@ const createFileStasher = (input) => {
...
-                      contentDisposition.parse(cd).parameters.filename;
-                  }
-                }
-              } else {
-                throw new Error(
-                  'Cannot stash a Promise wrapped file of unknown type.'
-                );
-              }
+        const parseFinalResponse = (stream) => {
+          var newBufferStringStream = stream;
```
Comment (eliangcs) [inline packages/core/src/tools/create-file-stasher.js:109]:
```suggestion
          let newBufferStringStream = stream;
```

**Review by eliangcs — APPROVED** [review]
I confirmed it fixes the issue. I only have a suggested change on the `var` vs. `let` lining error.

The file stasher code is one of the spots that always take me a long time to understand. So really appreciate you took the time to read the code and fix it!

**Comment by nicolemichellethomas** [comment]
@eliangcs Awesome! I've committed the let -> var change you suggested :)

**Review by eliangcs — APPROVED** [review]
💯

**Pull request merged**